src/eng_back.c

```diff
@@ -941,6 +942,7 @@ static int ctx_ctrl_set_pin(ENGINE_CTX *ctx, const char *pin)
 	 * shall be returned and errno shall be set. */
 	ctx_destroy_pin(ctx);
 	ctx->pin = OPENSSL_strdup(pin);
+	ctx->forced_pin = 1;
```

Comment:
nit: I would probably move this line after the allocation check or the length count to make the code easier to read and keep the pieces that belong together together.

Reply:
Done